### DISCUSSION TITLE
Keycloak Database support for reader and writer endpoints.

### DISCUSSION THREAD

**Discussion by sabinayakc** [keycloak/keycloak#33486] [Q&A] [answered]
We would like to connect keycloak to a reader and writer endpoints for the external database.
We currently are using AWS Aurora cluster with one reader and one writer endpoints.

How do we configure keycloak to use the reader and writer endpoints? Is this achievable?

**Comment by sabinayakc** [comment]
Looks like I need to check out the AWS JDBC driver?

https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/examples/AWSDriverExample/src/main/java/software/amazon/ReadWriteSplittingPostgresExample.java

**Reply by codespearhead** [reply]
Simply provide the database JDBC URL and the credentials for the writer endpoint to Keycloak, as you would for a regular database, as suggested in [1].

Although Keycloak doesn't support separating read and write operations at the application level, most of its database interactions require a connection to the writer endpoint, so it makes sense to keep it that way for simplicity's sake.

[1] https://www.keycloak.org/high-availability/deploy-aurora-multi-az#_deploying_keycloak

**Reply by sabinayakc** [reply]
The JDBC url mentions the `aws-wrapper`. Do I need to do install the aws jdbc driver using a custom image?

**Reply by codespearhead** [reply, marked as answer]
Yes, as explained in the documentation [1][2].

[1] https://www.keycloak.org/server/db#preparing-keycloak-for-amazon-aurora-postgresql
[2] https://github.com/keycloak/keycloak/blob/98a6ddead0d7121470f5d17be635e1f4dbf3058e/docs/guides/server/db.adoc?plain=1#L244
